tests/helpers/testers.py

```diff
@@ -400,6 +400,21 @@ def run_precision_test_gpu(
             metric_module(**metric_args), partial(metric_functional, **metric_args), preds, target, device="cuda"
         )
 
+    def test_differentiability(
+        self, 
+        preds: torch.Tensor, 
+        target: torch.Tensor,
+        metric_module: Metric,
+    ):
+        """Test if a metric is differentiable or not
+        Args:
+            preds: torch tensor with predictions
+            target: torch tensor with targets
+            metric_module: the metric module to test
+        """
+        preds.requires_grad = True
+        out = metric_module(preds, target)
+        assert metric_module.is_differentiable == out.requires_grad
```

Comment:
Why would `metric_module.is_differentiable == out.requires_grad` ? 
I understand the design logic, but the Metric `differentiable` property should be independent of the tensors
`requires_grad` property @SkafteNicki ?

Reply:
If a metric is differentiable then input that requires grad should result in output that also requires grad.
If a metric is not differentiable then if we give it input that requires grad then the output will not have a grad.